src/duration.ts

```diff
@@ -86,134 +87,214 @@ export class Duration {
 
 export function applyDuration(date: Date | number, duration: Duration): Date {
   const r = new Date(date)
-  r.setFullYear(r.getFullYear() + duration.years)
-  r.setMonth(r.getMonth() + duration.months)
-  r.setDate(r.getDate() + duration.weeks * 7 + duration.days)
-  r.setHours(r.getHours() + duration.hours)
-  r.setMinutes(r.getMinutes() + duration.minutes)
-  r.setSeconds(r.getSeconds() + duration.seconds)
+  const actions: Array<() => void> = [
```

Comment:
I can't see why we need an array here. Because of the `if`? Having an array of functions and reversing them seems costly compared to duplicating the small number of lines twice in an `if/else`

Reply:
If the duration to be applied is negative, the order of the operations will be reversed. This is so that adding a duration then subtracting the same duration always gives the original date, which was not the case with the old implementation.